pulp_gem/app/tasks/publishing.py

```diff
@@ -18,6 +19,25 @@
 log = logging.getLogger(__name__)
 
 
+index_template = """<!DOCTYPE html>
```

Comment:
So this generated index is basically replacing the on the fly index of the content app?

Reply:
Correct. When `gem` tries to install it does a `HEAD` request to the base_url and the content-app buckles when the repo has millions of content units, like when one has done a full sync of rubygems.org. Pre-computing the index speeds up the install tremendously.